lib/msal-common/src/request/CommonClientCredentialRequest.ts

```diff
@@ -4,14 +4,17 @@
  */
 
 import { BaseAuthRequest } from "./BaseAuthRequest";
+import { AzureRegion } from "../authority/AzureRegion";
 
 /**
  * CommonClientCredentialRequest
- * - scopes                  - Array of scopes the application is requesting access to.
- * - authority               - URL of the authority, the security token service (STS) from which MSAL will acquire tokens.
- * - correlationId           - Unique GUID set per request to trace a request end-to-end for telemetry purposes.
- * - skipCache               - Skip token cache lookup and force request to authority to get a a new token. Defaults to false.
+ * - scopes                             - Array of scopes the application is requesting access to.
+ * - authority                          - URL of the authority, the security token service (STS) from which MSAL will acquire tokens.
+ * - correlationId                      - Unique GUID set per request to trace a request end-to-end for telemetry purposes.
+ * - skipCache                          - Skip token cache lookup and force request to authority to get a a new token. Defaults to false.
+ * - preferredAzureRegionOptions        - Options of the user's preferred azure region
  */
 export type CommonClientCredentialRequest = BaseAuthRequest & {
     skipCache?: boolean;
+    preferredAzureRegion?: AzureRegion;
```

Comment:
Bogdan's comment is to make this common across ConfidentialClients. I do think we can add this for all requests separately when the use cases are supported, can you please check once though?

Reply:
I don't understand the ask here, could you shed more light on it?

Reply:
Addressed.